pkg/gameservers/gameservers.go

```diff
@@ -59,7 +78,7 @@ func address(node *corev1.Node) (string, error) {
 // applyGameServerAddressAndPort gathers the address and port details from the node and pod
 // and applies them to the GameServer that is passed in, and returns it.
 func applyGameServerAddressAndPort(gs *agonesv1.GameServer, node *corev1.Node, pod *corev1.Pod) (*agonesv1.GameServer, error) {
-	addr, err := address(node)
+	addr, err := address(node, runtime.FeatureEnabled(runtime.NodeExternalDNS))
```

Comment:
I would be tempted to keep the signature of `address(...)` the same and pull runtime.FeatureEnabled(runtime.NodeExternalDNS) into the function.

That way it's all encapsulated in one place - and also easier to adjust as we go from alpha :arrow_right: beta :arrow_right: to stable.

Also means less changes across the entire PR too.

Reply:
I thought the function itself would be more valuable if it were pure, but on this scale, it sounded like my preference.
Since we are not particular about it, we will try to keep the signature of the address function.

Reply:
Thanks! It also makes it consistent with other feature flag implementations.